mypy_django_plugin/transformers/models.py

```diff
@@ -250,61 +256,63 @@ def run_with_model_cls(self, model_cls: Type[Model]) -> None:
-                )
-                if not self.has_any_parametrized_manager_as_base(manager_info) or is_dynamically_generated:
-                    continue
+                continue
 
-                custom_model_manager_name = manager.model.__name__ + "_" + manager_class_name
-                try:
-                    custom_manager_type = self.create_new_model_parametrized_manager(
-                        custom_model_manager_name, base_manager_info=manager_info
-                    )
-                except helpers.IncompleteDefnException:
-                    continue
+            if self._is_declared_inside_class(manager_name):
+                continue
 
-                self.add_new_node_to_model_class(manager_name, custom_manager_type)
-
-        if incomplete_manager_defs and not self.api.final_iteration:
-            #  Unless we're on the final round, see if another round could figure out all manager types
-            raise helpers.IncompleteDefnException()
-        elif self.api.final_iteration:
-            for manager_name in incomplete_manager_defs:
-                # We act graceful and set the type as the bare minimum we know of
-                # (Django's default) before finishing. And emit an error, to allow for
-                # ignoring a more specialised manager not being resolved while still
-                # setting _some_ type
-                django_manager_info = self.lookup_typeinfo(fullnames.MANAGER_CLASS_FULLNAME)
-                assert (
-                    django_manager_info is not None
-                ), f"Type info for Django's {fullnames.MANAGER_CLASS_FULLNAME} missing"
-                self.add_new_node_to_model_class(
-                    manager_name, Instance(django_manager_info, [Instance(self.model_classdef.info, [])])
+            if not self._should_subclass_manager(manager_info):
+                self.add_new_node_to_model_class(manager_name, self.model_parametrize(manager_info))
+                continue
+
+            custom_manager_name = f"{manager.model.__name__}_{manager_class_name}"
+            try:
+                custom_manager_type = self.create_new_model_parametrized_manager(
+                    custom_manager_name, base_manager_info=manager_info
                 )
-                # Find expression for e.g. `objects = SomeManager()`
-                manager_expr = [
+            except helpers.IncompleteDefnException:
+                continue
+
+            self.add_new_node_to_model_class(manager_name, custom_manager_type)
+
+        if incomplete_manager_defs:
+            if not self.api.final_iteration:
+                # Unless we're on the final round, see if another round could figure out
+                # all manager types
+                raise helpers.IncompleteDefnException()
+            else:
+                self._report_incomplete_defs(incomplete_manager_defs)
+
+    def _report_incomplete_defs(self, incomplete_defs: Set[str]) -> None:
```

Comment:
I wouldn't really mind if `_report_incomplete_defs` is inlined instead of branched out in a separate method. But perhaps that's just me.

```suggestion
        if incomplete_manager_defs and not self.api.final_iteration:
            # Unless we're on the final round, see if another round could figure out
            # all manager types
            raise helpers.IncompleteDefnException()
        elif self.api.final_iteration:
```

Reply:
IMO, `run_with_model_cls` grows too large, and this part (>10 lines) can be extracted logically and with just a single argument. Too long methods are harder to debug (I had to put 5 breakpoints to `AddRelatedManagers.run_with_model_cls` just to find out where things go wrong working on another issue, it would be easier with shorter methods). I'd rather leave this extracted.

Reply:
Alright, sure, that makes sense